### ISSUE TITLE
adding java.nio.file.attribute.PosixFileAttributes ?

### ISSUE THREAD

**Issue by ryantate** [babashka/babashka#1776]
I've got a CLI script that loads POSIX and Basic file attributes in bulk with `java.nio.file.attribute.PosixFileAttributes` - this allows me to get the file permissions of an existing file along with the other (Basic) attributes .

As far as I know, this is the only way to read these permissions from an existing file. You provide `java.nio.file.attribute.PosixFilePermissions`, but without the POSIX reader class this is only good for either getting permissions from a shell string (ala `ls -l`) or setting permissions for a file you're creating.

(I'm specifically trying to determine when a file has `PosixFilePermission/OTHERS_READ` set, so I can't just use `Files/isReadable`, fyi.)

Would you be open to adding this class to babashka? If so, what would the next steps be - would a PR be helpful?

Thanks for considering.

**Comment by borkdude** [comment]
A PR for both issues would be welcome, preferably with a test in babashka.interop-test

**Comment by ryantate** [comment]
Currently it looks like one of your dependencies (`timbre`) is causing `script/test` to throw an error:

```
user@dev:~/Documents/clojure/babashka$ script/test
openjdk version "21.0.5" 2024-10-15
OpenJDK Runtime Environment (Red_Hat-21.0.5.0.11-1) (build 21.0.5+11)
OpenJDK 64-Bit Server VM (Red_Hat-21.0.5.0.11-1) (build 21.0.5+11, mixed mode, sharing)
running tests part 1
Composite profiles containing maps are strongly recommended against. 
Support will be removed in future versions of Leiningen due to subtle 
unexpected behavior. Move the map from the composite profile to its own 
top-level named profile to avoid issues.
 [:feature/xml :feature/lanterna :feature/yaml :feature/postgresql :feature/hsqldb :feature/csv :feature/transit :feature/datascript :feature/httpkit-client :feature/httpkit-server :feature/core-match :feature/hiccup :feature/test-check :feature/spec-alpha :feature/selmer :feature/logging :feature/priority-map :feature/rrb-vector {:dependencies ([borkdude/rewrite-edn "0.4.6"] [com.clojure-goes-fast/clj-async-profiler "0.5.0"] [com.opentable.components/otj-pg-embedded "0.13.3"] [nubank/matcher-combinators "3.6.0"])}]
Composite profiles containing maps are strongly recommended against. 
Support will be removed in future versions of Leiningen due to subtle 
unexpected behavior. Move the map from the composite profile to its own 
top-level named profile to avoid issues.
 [:feature/jdbc {:dependencies ([org.postgresql/postgresql "42.2.18"])}]
Composite profiles containing maps are strongly recommended against. 
Support will be removed in future versions of Leiningen due to subtle 
unexpected behavior. Move the map from the composite profile to its own 
top-level named profile to avoid issues.
 [:feature/jdbc {:dependencies ([org.hsqldb/hsqldb "2.5.1"])}]
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Syntax error compiling at (taoensso/timbre/appenders/core.cljc:50:17).
No such var: enc/println

Full report at:
/tmp/clojure-17291281772378040269.edn
Subprocess failed (exit code: 1)
```
I've got submodules checked out, lein installed, and so forth, but apologies if I've missed some step. 

Full report attached
[clojure-17291281772378040269.edn.txt](https://github.com/user-attachments/files/18199582/clojure-17291281772378040269.edn.txt)

**Comment by borkdude** [comment]
What leiningen version do you have?

**Comment by borkdude** [comment]
I'm on 

```
$ lein --version
Leiningen 2.9.10 on Java 23 Java HotSpot(TM) 64-Bit Server VM
```

but perhaps lein changed something?

**Comment by borkdude** [comment]
I've tried with a newer version of leiningen and can't reproduce the failures (I do get the warnings about the profiles). Perhaps there is some garbage in your `target` folder or so?

```
rm -rf target
```

Are you able to execute

```
lein do clean, test :only babashka.interop-test
```

?

**Comment by borkdude** [comment]
I don't want this to be a blocker to solve the issue btw, so if you could just provide a snippet that could function as a test, I could also just solve the issue for you.

**Comment by ryantate** [comment]
I'm sorry, this is entirely my fault.

I forgot I had a `~/.lein/profiles.clj`; it has been a long time since I used `lein`. Anyway in this file were dependencies on an older tufte and criterium. This Tufte version depended on a quite old encore version (3.24 vs 3.127 from timbre). 

I presume the dependency was resolved to an older encore without the println var.

Now the tests run fine, except for one failure due to sefmentation fault in the pod for go-sqlite, which does not concern my patch, so it's fine. `interop-test` passes fine.

Sorry again.
